airflow/executors/knative_executor.py

```diff
@@ -0,0 +1,186 @@
+from airflow.utils.state import State
+from airflow.exceptions import AirflowException
+import asyncio
+import datetime
+import aiohttp
+import functools
+from airflow.contrib.executors.kubernetes_executor import KubernetesExecutor
+
+
+async def make_request_async(
+    task_id,
+    dag_id,
+    execution_date,
+    host,
+    log,
+    host_header=None,
+) -> aiohttp.ClientResponse:
+    req = "http://" + host + "/run"
+
+    date = int(datetime.datetime.timestamp(execution_date))
+    params = {
+        "task_id": task_id,
+        "dag_id": dag_id,
+        "execution_date": date,
+    }
+    headers = {}
+    if host_header:
+        headers["Host"] = host_header
+    timeout = aiohttp.ClientTimeout(total=60000)
+    async with aiohttp.ClientSession(timeout=timeout) as session:
+        async with session.get(url=req, params=params, headers=headers) as resp:
+            log.info(resp.status)
+            log.info(await resp.text())
+            return resp
+
+
+class KnativeRequestLoop(multiprocessing.Process, LoggingMixin):
+    def __init__(self,
+                 task_pipe,
+                 result_pipe,
+                 pop_running_pipe,
+                 host_header,
+                 host,
+                 ):
+
+        super().__init__()
+        self.host = host
+        self.task_pipe = task_pipe
+        self.result_pipe = result_pipe
+        self.host_header = host_header
+        self.pop_running_pipe = pop_running_pipe
+
+    def recieve_and_execute(self, loop):
+        key = self.task_pipe.recv()
+        loop.create_task(self.execute_work(key=key))
+
+    def run(self):
+        loop = asyncio.get_event_loop()
+        loop.add_reader(fd=self.task_pipe, callback=functools.partial(self.recieve_and_execute, loop))
+        loop.run_forever()
+
+    async def execute_work(self, key):
+        """
+        Executes command received and stores result state in queue.
+        :param task_instance:
+        :param key: the key to identify the TI
+        :type key: tuple(dag_id, task_id, execution_date)
+        :param command: the command to execute
+        :type command: str
+        """
+        if key is None:
+            return
+        self.log.info("%s running %s", self.__class__.__name__, key)
+        (dag_id, task_id, execution_date, try_number) = key
+        try:
+            future = make_request_async(
+                task_id,
+                dag_id,
+                execution_date,
+                host=self.host, log=self.log, host_header=self.host_header)
+            resp = await future
+            if resp.status != 200:
+                state = State.FAILED
+                self.log.error("Failed to execute task %s.", str(resp.text))
+                self.result_pipe.send((key, state))
+            else:
+                self.log.info("assuming task success")
+                # self.result_queue.put((key, None))
+                self.pop_running_pipe.send(key)
```

Comment:
What was the your thinking for having a separate pipe for this - `self.result_queue.put((key, None))` or `self.result_queue.put((key, State.SUCCESS))` seems like it would work?